zap/src/test/java/org/zaproxy/zap/extension/pscan/PassiveScanDataUnitTest.java

```diff
@@ -0,0 +1,427 @@
+        // Given
+        CustomPage.Type type = CustomPage.Type.NOTFOUND_404;
+        HttpMessage msg = createMessage();
+        given(context.isCustomPageWithFallback(msg, type)).willReturn(true);
+        given(context.isCustomPageWithFallback(msg, CustomPage.Type.OK_200)).willReturn(false);
+        given(context.isCustomPageWithFallback(msg, CustomPage.Type.ERROR_500)).willReturn(false);
+        given(session.getContextsForUrl(msg.getRequestHeader().getURI().toString()))
+                .willReturn(asList(context));
+        PassiveScanData psd = new PassiveScanData(msg);
+        // When
+        boolean result = psd.isPage404(msg);
+        // Then
+        assertThat(result, is(equalTo(true)));
+        verify(context).isCustomPageWithFallback(msg, CustomPage.Type.OK_200);
+        verify(context).isCustomPageWithFallback(msg, type);
+    }
+
+    @Test
+    public void shouldCheckPage500WithContext() {
+        // Given
+        CustomPage.Type type = CustomPage.Type.ERROR_500;
+        HttpMessage msg = createMessage();
+        given(context.isCustomPageWithFallback(msg, type)).willReturn(true);
+        given(context.isCustomPageWithFallback(msg, CustomPage.Type.OK_200)).willReturn(false);
+        given(context.isCustomPageWithFallback(msg, CustomPage.Type.NOTFOUND_404))
+                .willReturn(false);
+        given(session.getContextsForUrl(msg.getRequestHeader().getURI().toString()))
+                .willReturn(asList(context));
+        PassiveScanData psd = new PassiveScanData(msg);
+        // When
+        boolean result = psd.isPage500(msg);
+        // Then
+        assertThat(result, is(equalTo(true)));
+        verify(context).isCustomPageWithFallback(msg, CustomPage.Type.OK_200);
+        verify(context).isCustomPageWithFallback(msg, type);
+    }
+
+    @Test
+    public void shouldCheckPageOtherWithContext() {
+        // Given
+        CustomPage.Type type = CustomPage.Type.OTHER;
+        HttpMessage msg = createMessage();
+        given(context.isCustomPageWithFallback(msg, type)).willReturn(true);
+        given(session.getContextsForUrl(msg.getRequestHeader().getURI().toString()))
+                .willReturn(asList(context));
+        PassiveScanData psd = new PassiveScanData(msg);
+        // When
+        boolean result = psd.isPageOther(msg);
+        // Then
+        assertThat(result, is(equalTo(true)));
+        verify(context).isCustomPageWithFallback(msg, type);
+    }
+
+    @Test
+    public void isClientErrorShouldReturnTrueIfStatusCodeMatches() {
+        // Given
+        HttpMessage msg = createMessage();
+        msg.getResponseHeader().setStatusCode(403);
+        PassiveScanData psd = new PassiveScanData(msg);
+        // When
+        boolean result = psd.isClientError(msg);
+        // Then
+        assertThat(result, is(equalTo(true)));
+        verifyNoInteractions(context);
+    }
+
+    @Test
+    @MockitoSettings(strictness = Strictness.LENIENT)
```

Comment:
It would be better to setup the behaviour for 200/500 rather than setting it lenient.